.svglintrc.js

```diff
@@ -129,6 +132,34 @@ module.exports = {
               }
             }
           },
+          function(reporter, $, ast) {
+            reporter.name = "icon-precision";
+
+            const iconPath = $.find("path").attr("d");
+            const { segments } = parsePath(iconPath);
+            const segmentParts = segments.flat().filter((num) => (typeof num === 'number'));
+
+            const countDecimals = (num) => {
+              if (num && num % 1) {
+                let [base, op, trail] = num.toExponential().split(/e([+-])/);
+                let elen = parseInt(trail, 10);
+                let idx = base.indexOf('.');
+                return idx == -1 ? elen : base.length - idx - 1 + (op === '+' ? -elen : elen);
+              }
+              return 0;
+            };
+            const precisionArray = segmentParts.map(countDecimals);
+            const precisionAverage = precisionArray && precisionArray.length > 0 ?
+              Math.round(precisionArray.reduce((prev, curr) => prev + curr) / precisionArray.length) :
+              0;
```

Comment:
Why do you need this? :thinking:

```suggestion
            const precisionAverage = Math.round(precisionArray.reduce((prev, curr) => prev + curr) / precisionArray.length);
```

Reply:
It surely is a edge case, but it is to prevent errors if the path is invalid. With the proposal, it gives an error like the following:
![image](https://user-images.githubusercontent.com/12817388/98849948-01a4d600-2422-11eb-96f0-840dd06392cb.png)
The invalid path is already catch by another reporter and the error coming from the precision is not really useful for the end user.

If I keep the code as-is, this is what I get with an invalid path:
![image](https://user-images.githubusercontent.com/12817388/98850136-4b8dbc00-2422-11eb-943a-0f2c29efe411.png)
Not exactly the truth, but better than a exception I think!

What do you think?

Reply:
I see, perfect! :+1: